src/Microsoft.OpenApi.Kiota.ApiDescription.Client/Microsoft.OpenApi.Kiota.ApiDescription.Client.targets

```diff
@@ -17,5 +20,25 @@
     <Exec Command="%(CurrentOpenApiReference.Command)" LogStandardErrorAsError="true" />
   </Target>
 
+  <!-- OpenApiReference support for TypeScript -->
+
+  <Target Name="GenerateKiotaTypeScript">
+    <ItemGroup>
+      <!-- @(CurrentOpenApiReference) item group will never contain more than one item. -->
+      <CurrentOpenApiReference>
+        <Command>$(_KiotaCommand) -d %(FullPath) -o %(OutputPath) -l TypeScript  -c %(ClassName) -n %(Namespace)</Command>
+      </CurrentOpenApiReference>
+    </ItemGroup>
+
+    <Message Importance="high" Text="%0AGenerateKiotaTypeScript:" />
+    <Message Importance="high" Text="  %(CurrentOpenApiReference.Command)" />
+
+    <Exec Command="%(CurrentOpenApiReference.Command)" LogStandardErrorAsError="true" />
+    <Exec ContinueOnError="True" Command='npm i @microsoft/kiota-abstractions @microsoft/kiota-serialization-json @microsoft/kiota-serialization-text @microsoft/kiota-authentication-azure @microsoft/kiota-http-fetch -S'>
```

Comment:
might need to execute in $(SpaRoot) where the package.json lives for react/angular + aspnet. For aspnet plain, the generate file is dropped in obj, the TypeScript targets try to build it but people are expected to serve/copy it themselves